src/cpp/fastdds/xtypes/dynamic_types/DynamicTypeImpl.cpp

```diff
@@ -0,0 +1,217 @@
+}
+
+ReturnCode_t DynamicTypeImpl::get_member_by_name(
+        traits<DynamicTypeMember>::ref_type& member,
+        const ObjectName& name) noexcept
+{
+    auto it = member_by_name_.find(name);
+
+    if (member_by_name_.end() == it)
+    {
+        return RETCODE_BAD_PARAMETER;
+    }
+
+    member = it->second;
+    return RETCODE_OK;
+}
+
+ReturnCode_t DynamicTypeImpl::get_all_members_by_name(
+        DynamicTypeMembersByName& member) noexcept
+{
+    member = member_by_name_;
+    return RETCODE_OK;
+}
+
+ReturnCode_t DynamicTypeImpl::get_member(
+        traits<DynamicTypeMember>::ref_type& member,
+        MemberId id) noexcept
+{
+    auto it = member_.find(id);
+
+    if (member_.end() == it)
+    {
+        return RETCODE_BAD_PARAMETER;
+    }
+
+    member = it->second;
+    return RETCODE_OK;
+}
+
+ReturnCode_t DynamicTypeImpl::get_all_members(
+        DynamicTypeMembersById& member) noexcept
+{
+    member = member_;
+    return RETCODE_OK;
+}
+
+uint32_t DynamicTypeImpl::get_member_count() noexcept
+{
+    return static_cast<uint32_t>(members_.size());
+}
+
+ReturnCode_t DynamicTypeImpl::get_member_by_index(
+        traits<DynamicTypeMember>::ref_type& member,
+        uint32_t index) noexcept
+{
+    if (index >= members_.size())
+    {
+        return RETCODE_BAD_PARAMETER;
+    }
+
+    member = members_.at(index);
+    return RETCODE_OK;
+}
+
+uint32_t DynamicTypeImpl::get_annotation_count() noexcept
+{
+    return static_cast<uint32_t>(annotation_.size());
+}
+
+ReturnCode_t DynamicTypeImpl::get_annotation(
+        traits<AnnotationDescriptor>::ref_type& descriptor,
+        uint32_t idx) noexcept
+{
+    if (!descriptor || idx >= annotation_.size())
+    {
+        return RETCODE_BAD_PARAMETER;
+    }
+
+    traits<AnnotationDescriptor>::narrow<AnnotationDescriptorImpl>(descriptor)->copy_from(annotation_.at(idx));
+    return RETCODE_OK;
+}
+
+uint32_t DynamicTypeImpl::get_verbatim_text_count() noexcept
+{
+    return static_cast<uint32_t>(verbatim_.size());
+}
+
+ReturnCode_t DynamicTypeImpl::get_verbatim_text(
+        traits<VerbatimTextDescriptor>::ref_type& descriptor,
+        uint32_t idx) noexcept
+{
+    if (!descriptor || idx >= verbatim_.size())
+    {
+        return RETCODE_BAD_PARAMETER;
+    }
+
+    traits<VerbatimTextDescriptor>::narrow<VerbatimTextDescriptorImpl>(descriptor)->copy_from(verbatim_.at(idx));
+    return RETCODE_OK;
+}
+
+bool DynamicTypeImpl::equals(
+        traits<DynamicType>::ref_type other) noexcept
+{
+    bool ret_value = true;
+    auto impl = traits<DynamicType>::narrow<DynamicTypeImpl>(other);
+
+    ret_value &= type_descriptor_.equals(impl->type_descriptor_);
+    if (ret_value)
+    {
+        ret_value &= annotation_.size() == impl->annotation_.size();
+        if (ret_value)
+        {
+            for (size_t count {0}; ret_value && count < annotation_.size(); ++count)
+            {
+                ret_value &= annotation_.at(count).equals(impl->annotation_.at(count));
```

Comment:
annotation_ is a vector. Consequently, it could be that the annotations are the same but are included in different order, but the order does not make the DynamicType not equal.

Reply:
Done in c8b1745ba